components/molecule/phoneInput/src/index.js

```diff
@@ -0,0 +1,161 @@
+import {useEffect, useRef, useState} from 'react'
+
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import flagIcons from 'rendered-country-flags'
+
+import AtomIcon, {
+  ATOM_ICON_COLORS,
+  ATOM_ICON_SIZES
+} from '@s-ui/react-atom-icon'
+import AtomInput from '@s-ui/react-atom-input'
+import {TYPES} from '@s-ui/react-atom-input/lib/config.js'
+import MoleculeDropdownList, {
+  moleculeDropdownListDesigns
+} from '@s-ui/react-molecule-dropdown-list'
+import MoleculeDropdownOption from '@s-ui/react-molecule-dropdown-option'
+
+import {phoneValidationType} from './settings.js'
+
+const BASE_CLASS = 'sui-MoleculePhoneInput'
+
+export default function MoleculePhoneInput({
+  dropdownIcon,
+  value = '',
+  prefixes = [],
+  onChange,
+  placeholder,
+  hasError,
+  initialSelectedPrefix = {},
+  type = phoneValidationType.DEFAULT
+}) {
+  const [showDropdown, setShowDropdown] = useState(false)
+  const [selectedPrefix, setSelectedPrefix] = useState(initialSelectedPrefix)
+  const modalRef = useRef(null)
+  const inputPrefixRef = useRef(null)
+
+  const baseClass = cx(
+    {
+      [`splitted`]: type === phoneValidationType.SPLITTED,
+      [`${BASE_CLASS}--error`]: hasError
+    },
+    BASE_CLASS
+  )
+
+  // Close dropdown when click outside
+  useEffect(() => {
+    const handleClickOutside = event => {
+      if (
+        modalRef.current &&
+        !modalRef.current.contains(event.target) &&
+        !inputPrefixRef.current.contains(event.target)
+      ) {
+        setShowDropdown(false)
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+    }
+  }, [modalRef])
```

Comment:
`modalRef` shouldn't change during re-renders so I think is not needed here

Reply:
Alright, gonna change it then